operator/cmd/mesh/uninstall_test.go

```diff
@@ -27,22 +27,7 @@ var (
 	logOpts = log.DefaultOptions()
 )
 
-func TestUninstallNonEmptyRevision(t *testing.T) {
-	t.Skip("https://github.com/istio/istio/issues/30182")
-	g := gomega.NewWithT(t)
-	args := []string{"--revision", "rev"}
-	uninstallCmd := UninstallCmd(logOpts)
-	uninstallCmd.SetArgs(args)
-	var out bytes.Buffer
-	uninstallCmd.SetOut(&out)
-	uninstallCmd.SetErr(&out)
-
-	err := uninstallCmd.Execute()
-	g.Expect(err).To(gomega.HaveOccurred())
-}
-
 func TestUninstallEmptyRevision(t *testing.T) {
```

Comment:
This is still trying to use the real cluster, is it? or at the very least, if it fails due to some bug, it would start to use the real cluster.

Its not great if a unit test failing causes my cluster to get uninstalled..

Reply:
makes sense, I would remove these remaining parts as well

Reply:
+1 to remove the remaining parts.
Also install tests in https://github.com/istio/istio/blob/master/operator/cmd/mesh/install_test.go

Reply:
The install command has a dryrun option so that would be safe, another option is to add dryrun option to uninstall command as well, I think that would also help improve ux experience

Reply:
That does not make it safe, it will still reach the cluster. A unit test should *never* make any calls (even reads) from the users kubeconfig

Reply:
ok I just removed them